commit a66193f585ebc3c63fb3b6c767bd20191b5820a2
Author: Samuel Mendoza-Jonas <sam@mendozajonas.com>
Date:   Wed Aug 22 14:57:44 2018 +1000

    net/ncsi: Fixup .dumpit message flags and ID check in Netlink handler
    
    [ Upstream commit 3d0371b313b84ba7c16ebf2526a7a34f1c57b19e ]
    
    The ncsi_pkg_info_all_nl() .dumpit handler is missing the NLM_F_MULTI
    flag, causing additional package information after the first to be lost.
    Also fixup a sanity check in ncsi_write_package_info() to reject out of
    range package IDs.
    
    Signed-off-by: Samuel Mendoza-Jonas <sam@mendozajonas.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/net/ncsi/ncsi-netlink.c b/net/ncsi/ncsi-netlink.c
index 82e6edf9c5d9..45f33d6dedf7 100644
--- a/net/ncsi/ncsi-netlink.c
+++ b/net/ncsi/ncsi-netlink.c
@@ -100,7 +100,7 @@ static int ncsi_write_package_info(struct sk_buff *skb,
 	bool found;
 	int rc;
 
-	if (id > ndp->package_num) {
+	if (id > ndp->package_num - 1) {
 		netdev_info(ndp->ndev.dev, "NCSI: No package with id %u\n", id);
 		return -ENODEV;
 	}
@@ -240,7 +240,7 @@ static int ncsi_pkg_info_all_nl(struct sk_buff *skb,
 		return 0; /* done */
 
 	hdr = genlmsg_put(skb, NETLINK_CB(cb->skb).portid, cb->nlh->nlmsg_seq,
-			  &ncsi_genl_family, 0,  NCSI_CMD_PKG_INFO);
+			  &ncsi_genl_family, NLM_F_MULTI,  NCSI_CMD_PKG_INFO);
 	if (!hdr) {
 		rc = -EMSGSIZE;
 		goto err;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1277_a66193f585ebc3c63fb3b6c767bd20191b5820a2.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1277_a66193f585ebc3c63fb3b6c767bd20191b5820a2.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
